Filters/Average.cs

```diff
@@ -3,48 +3,61 @@
 
 namespace Scarlet.Filters
 {
-    /// <summary> The Average filter is intended for use as an average-gathering system, using a rolling average with "roll-length" <c>FilterCount</c>.</summary>
+    /// <summary> 
+    /// The Average filter is intended for use as an average-gathering system, using a rolling average 
+    /// with "roll-length" <c>FilterCount</c> or continuous average.
+    /// </summary>
     /// <remarks>
-    /// Implementation Details:
-    /// 
-    /// *Construct Average filter given a rolling
-    ///  filter length, <c>FilterCount</c>
-    /// 
-    /// *Iteratively add values into the filter
-    ///  using <c>Feed(T Input)</c>
-    /// 
-    /// *Get the filter output by calling
-    ///  <c>YourFilterInstance.Output</c>
+    /// Intended Usage:
+    /// * Construct Average filter given a rolling
+    ///   filter length, <see cref="FilterCount">FilterCount</see>. Setting
+    ///   FilterCount to null puts the filter into
+    ///   continuous mode.
+    /// * Iteratively add values into the filter
+    ///   using <see cref="Feed(T)">Feed()</see>
+    /// * Get the filter output by calling
+    ///   <see cref="GetOutput">YourFilterInstance.GetOutput()</see>
     /// </remarks>
     /// <typeparam name="T"> A type, which must be a numeric. </typeparam>
     public class Average<T> : IFilter<T> where T : IComparable
     {
-
+        private readonly int? FilterCount;
         private T Output; // Filter output
         private dynamic[] AverageArray; // Stored average array
-        private dynamic CurSum;         // Current sum of the average array 
-        private int FilterCount,        // Size of the average array
-                    Index,              // Current index of the average array
-                    Iterations;         // Number of iterations in the filter
+        private dynamic CurSum; // Current sum of the average array 
+        private int Index, // The current index of the average filter (when in roll mode)
```

Comment:
If you're going to put variables on seperate lines, then copy the `private int` as well.